### ISSUE TITLE
Can we convert pragma "no doc" to an attribute?

### ISSUE THREAD

**Issue by arezaii** [chapel-lang/chapel#21485]
With potential changes to attributes that would come with https://github.com/chapel-lang/chapel/pull/21425, is it acceptable to deprecate and replace `pragma "no doc"` with an attribute that carries the same meaning?

I would like to propose using the attribute toolspace for `chpldoc`, so the full attribute would be `@chpldoc.nodoc()` or `@chpldoc.nodoc` depending on if we end up requiring parens on attributes or not (see https://github.com/chapel-lang/chapel/issues/20994#issuecomment-1409735935).

I believe that we identified no-doc as a candidate for attributes in https://github.com/chapel-lang/chapel/issues/14141#issuecomment-1273791742. 

Are there currently any reservations about making a change like this?

edit to add code examples:

Today, we mark Chapel code that we don't want documented like this:
```chapel
pragma "no doc"
proc myUndocumentedFunction() {}
```

With the changes in https://github.com/chapel-lang/chapel/pull/21425 we could convert these from pragmas to attributes. 
In this issue I am proposing we use an attribute that is specific to the `chpldoc` tool, indicated with a property of attributes, the toolspace (a name and idea borrowed from Rust), like so:
```chapel
@chpldoc.nodoc()
proc myUndocumentedFunction() {}
```

**Comment by lydia-duncan** [comment]
I don't have any reservations about this and would welcome this change!

**Comment by bradcray** [comment]
I agree with making it an attribute one way or another.  I'd like to see a standalone issue about the use of hierarchical attributes / toolspaces, which I don't think has received much attention or broad discussion?  (though I'm still way behind, so if it did and was advertised/discussed, just point me in the right direction).

**Comment by arezaii** [comment]
see https://github.com/chapel-lang/chapel/issues/21493 for further discussion on toolspaces.

**Comment by lydia-duncan** [comment]
In looking for another issue, I discovered this was still open but I think it can be closed, right Ahmad?

**Comment by arezaii** [comment]
closing, as implemented in #21425. A future PR will deprecate the use of `pragma "no doc"` and point to the attribute as a replacement.
